src/display.cpp

```diff
@@ -1016,48 +1016,39 @@ std::pair<std::string, nc_color> display::vehicle_fuel_percent_text_color( const
     return std::make_pair( fuel_text, fuel_color );
 }
 
-std::vector<std::pair<std::string, nc_color>> display::bodypart_status_colors( const Character &u,
+// Return strings for all statuses affecting body part (bleeding, bitten, bandaged, etc.)
+static std::map<bodypart_status, nc_color> bodypart_status_colors( const Character &u,
         const bodypart_id &bp )
 {
-    // List of status strings and colors
-    std::vector<std::pair<std::string, nc_color>> ret;
+    std::map<bodypart_status, nc_color> ret;
+
     // Empty if no bodypart given
     if( bp == bodypart_str_id::NULL_ID() ) {
         return ret;
     }
 
-    const int bleed_intensity = u.get_effect_int( effect_bleed, bp );
-    const bool bleeding = bleed_intensity > 0;
-    const bool bitten = u.has_effect( effect_bite, bp.id() );
-    const bool infected = u.has_effect( effect_infected, bp.id() );
-    const bool broken = u.is_limb_broken( bp ) && bp->is_limb;
-    const bool splinted = u.worn_with_flag( json_flag_SPLINT,  bp );
-    const bool bandaged = u.has_effect( effect_bandaged,  bp.id() );
-    const bool disinfected = u.has_effect( effect_disinfected,  bp.id() );
-
     // Ailments
-    if( broken ) {
-        ret.emplace_back( std::make_pair( "broken", c_magenta ) );
+    if( u.has_effect( effect_bite, bp ) ) {
+        ret[bodypart_status::BITEN] = c_yellow;
     }
-    if( bitten ) {
-        ret.emplace_back( std::make_pair( "bitten", c_yellow ) );
+    if( u.has_effect( effect_infected, bp ) ) {
+        ret[bodypart_status::INFECTED] = c_pink;
     }
-    if( bleeding ) {
-        ret.emplace_back( std::make_pair( "bleeding",
-                                          colorize_bleeding_intensity( bleed_intensity ) ) );
+    if( u.is_limb_broken( bp ) ) {
+        ret[bodypart_status::BROKEN] = c_magenta;
     }
-    if( infected ) {
-        ret.emplace_back( std::make_pair( "infected", c_pink ) );
+    if( const int bleed = u.get_effect_int( effect_bleed, bp ) > 0 ) {
```

Comment:
I was on the fence about refactoring these, but stuff like this (assignment inside a condition) is partly why I chose the block of `const` declarations ahead of time - and to allow for the possibility of more complex `if` conditions later, combining several status checks.

I don't have strong feelings about the rest but I really dislike assignments inside conditions and would prefer to see it done another way.

Reply:
No problem! I restored the const declarations for status effects.